### PR TITLE
Only run needed tasks

### PR DESCRIPTION
**Pull request by jpa198**

### CODE REVIEW

File: e2e_tests/E2ETests.py [hunk 1]
```diff
@@ -385,6 +385,7 @@ def _run_auto(self, user: str, sleep_time: int = 10):
                         "Operational error while accessing database. "
                         "Retrying in {} seconds\n{}".format(sleep_time, ex)
                     )
+                    time.sleep(5)
```
Comment (jasonmotha) [inline e2e_tests/E2ETests.py]: Isn't this a parameter?

File: scripts/cybershake/auto_submit.py [hunk 2]
```diff
@@ -35,6 +35,10 @@
     const.ProcessType.HF: const.HPC.maui,
     const.ProcessType.BB: const.HPC.maui,
     const.ProcessType.IM_calculation: const.HPC.maui,
+    const.ProcessType.IM_plot: const.HPC.mahuika,
+    const.ProcessType.rrup: const.HPC.mahuika,
```
Comment (jasonmotha) [inline scripts/cybershake/auto_submit.py]: A cool extension would be to run things that are platform agnostic on whichever platform has a shorter queue. Or to saturate both queues.

File: scripts/cybershake/auto_submit.py [hunk 3]
```diff
@@ -608,6 +611,15 @@ def main(args, main_logger: Logger = workflow_logger.get_basic_logger()):
...
         args.n_runs = DEFAULT_N_RUNS
 
+    logger.debug("Processes to be run were: {}. Getting all required dependencies now.".format(args.tasks_to_run))
+    args.tasks_to_run = [const.ProcessType.get_by_name(proc) for proc in args.tasks_to_run]
+    for task in args.tasks_to_run:
+        logger.debug("Process {} in processes to be run, adding dependencies now.".format(task.str_value))
+        for proc_num in task.dependencies:
+            proc = const.ProcessType(proc_num)
+            if proc not in args.tasks_to_run:
+                args.tasks_to_run.append(proc)
```
Comment (jasonmotha) [inline scripts/cybershake/auto_submit.py]: I presume this cascades well? I can't explicity see that here but presume that adding to the iterating list gives that cascading.

File: scripts/management/MgmtDB.py [hunk 4]
```diff
@@ -96,14 +96,19 @@ def get_runnable_tasks(self, retry_max):
             Process.Verification.value,
         ]
 
+        if allowed_tasks is None:
+            allowed_tasks = list(const.ProcessType)
+        allowed_tasks = list(map(lambda x: str(x.value), allowed_tasks))
```
Comment (jasonmotha) [inline scripts/management/MgmtDB.py]: List comprehension.

File: scripts/management/MgmtDB.py [hunk 5]
```diff
@@ -96,14 +96,19 @@ def get_runnable_tasks(self, retry_max):
...
         with connect_db_ctx(self._db_file) as cur:
             db_tasks = cur.execute(
                 """SELECT proc_type, run_name, status_enum.state 
                           FROM status_enum, state 
                           WHERE state.status = status_enum.id
+                           AND proc_type IN ({})
                            AND (((status_enum.state = 'created' OR status_enum.state = 'failed')  
                                  AND state.retries <= ?)
-                            OR status_enum.state = 'completed')""",
+                            OR status_enum.state = 'completed')""".format(", ".join(allowed_tasks)),
```
Comment (jasonmotha) [inline scripts/management/MgmtDB.py]: Use prepared queries rather than string formatting.

File: scripts/cybershake/auto_submit.py (hunk 3 above)
Comment (jpa198) [reply scripts/cybershake/auto_submit.py]: Yep, adding to the end of the list means the for loop is dynamic and only ends when we stop adding things

File: scripts/cybershake/auto_submit.py (hunk 2 above)
Comment (jpa198) [reply scripts/cybershake/auto_submit.py]: Sounds like it should have its own task

Comment (jasonmotha) [reply scripts/cybershake/auto_submit.py]: Yeah. I'm just droning on.

**Pull request merged**